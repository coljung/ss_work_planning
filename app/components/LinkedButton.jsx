import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import { browserHistory } from 'react-router';

const LinkedButton = (props) => {
  const navigate = (to) => {
      browserHistory.push(to);
  }

  return (
      <Button onClick={navigate.bind(null, props.to)} {...props}>{props.children}</Button>
  );
}

LinkedButton.propTypes = {
    to: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]).isRequired,
};

export default LinkedButton;
