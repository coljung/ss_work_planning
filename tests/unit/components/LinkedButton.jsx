import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Button } from 'antd';
import LinkedButton from '../../../app/components/LinkedButton';

describe('LinkedButton', () => {
    it('should render correctly', () => {
        const output = shallow(
            <Button title="mockTitle" url="mockUrl" />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
