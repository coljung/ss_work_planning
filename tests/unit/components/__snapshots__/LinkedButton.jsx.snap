// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedButton should render correctly 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  title="mockTitle"
  type="button"
  url="mockUrl"
/>
`;
